src/backend/services/mining-api/src/routes.py

```diff
@@ -1,13 +1,59 @@
 from src import app, socketio
+from flask import request, jsonify
+from .MiningPool import MiningPool
+import sys, os
+import threading
+
+if os.environ.get('SERVICE_IN_DOCKER', False):
+    sys.path.append(os.path.abspath(os.path.join('..', '')))
+else:
+    sys.path.append(os.path.abspath(os.path.join('../..', '')))
+
+from shared import HttpCode, FailureReturnString
+
+
+ongoing_proof_for_id = None
+COINBASE_AMOUNT = 10
+
+def sendToConnectedClients(transaction):
+    # ToDo
+    print("To Be Sent to Miner", transaction)
+    try:
+        ongoing_proof_for_id = transaction["id"]
+        socketio.emit("find_proof", transaction)
+    except Exception as e:
+        print(e)
+
+transactions = MiningPool(sendToConnectedClients, True)
 
 @app.route("/")
-@app.route("/mining")
 def index():
     return "Hello From the Mining"
 
+@app.route("/queue", methods= ["POST"])
+def addDataToQueue():
+    data = request.get_json()
+
+    if data is None:
+        return jsonify(FailureReturnString.INCORRECT_PAYLOAD), HttpCode.BAD_REQUEST
+    
+    transactions.addToPool(data)
+
+    return jsonify(success=True), HttpCode.OK
+    
+
 @socketio.on('echo')
 def handle(message):
-    socketio.emit("response", message, callback=msg)
+    socketio.emit("response", message)
+
+@socketio.on('proof')
+def handle_proofs(message):
+    transactions.ready_to_mine()
+    # TODO if proof for wrong id is received compare it with the current ongoing proof
```

Comment:
Shouldn't be possible for a miner to send a proof for the wrong transaction as we will only have 1 mining session going at a time, and that mining will end server-side and client-side before another one begins